### PR TITLE
Support Search Attributes on Start workflow

### PR DESCRIPTION
**Pull request by vancexu**
https://github.com/uber/cadence-java-client/issues/347

### CODE REVIEW

File: src/main/java/com/uber/cadence/client/WorkflowOptions.java [hunk 1]
```diff
@@ -192,6 +196,12 @@ public Builder setMemo(Map<String, Object> memo) {
       return this;
     }
 
+    /** Specifies additional indexed information in result of list workflow. */
```
Comment (mfateev) [inline src/main/java/com/uber/cadence/client/WorkflowOptions.java]: Which types are allowed as values? Are they serialized in any way?

Comment (mfateev) [reply src/main/java/com/uber/cadence/client/WorkflowOptions.java]: The same question applies to memo.

File: src/main/java/com/uber/cadence/internal/sync/WorkflowStubImpl.java [hunk 2]
```diff
@@ -139,24 +139,24 @@ private StartWorkflowExecutionParameters getStartWorkflowExecutionParameters(
     }
     p.setInput(dataConverter.toData(args));
     p.setWorkflowType(new WorkflowType().setName(workflowType.get()));
-    p.setMemo(convertMemoFromObjectToBytes(o.getMemo()));
+    p.setMemo(convertMapFromObjectToBytes(o.getMemo()));
+    p.setSearchAttributes(convertMapFromObjectToBytes(o.getSearchAttributes()));
```
Comment (mfateev) [inline src/main/java/com/uber/cadence/internal/sync/WorkflowStubImpl.java]: I don't think you want JSON as search attribute value.

File: src/test/java/com/uber/cadence/client/WorkflowOptionsTest.java [hunk 3]
```diff
@@ -188,4 +196,10 @@ public void testInvalidCronScheduleAnnotation() throws NoSuchMethodException {
     memo.put("testKey", "testObject");
     return memo;
   }
+
+  private Map<String, Object> getTestSearchAttributes() {
+    Map<String, Object> searchAttr = new HashMap<>();
+    searchAttr.put("CustomKeywordField", "testKey");
```
Comment (mfateev) [inline src/test/java/com/uber/cadence/client/WorkflowOptionsTest.java]: May be more interesting type than string?

File: src/main/java/com/uber/cadence/client/WorkflowOptions.java (hunk 1 above)
Comment (vancexu) [reply src/main/java/com/uber/cadence/client/WorkflowOptions.java]: Comments updated

File: src/test/java/com/uber/cadence/client/WorkflowOptionsTest.java (hunk 3 above)
Comment (vancexu) [reply src/test/java/com/uber/cadence/client/WorkflowOptionsTest.java]: Added more types

File: src/main/java/com/uber/cadence/internal/sync/WorkflowStubImpl.java (hunk 2 above)
Comment (vancexu) [reply src/main/java/com/uber/cadence/internal/sync/WorkflowStubImpl.java]: You are right, for now client-server requires such Json encoding, but we can remove it. https://github.com/uber/cadence/issues/2433

**Pull request merged**